common/extent-tree-utils.c

```diff
@@ -285,3 +285,93 @@ int btrfs_record_file_extent(struct btrfs_trans_handle *trans,
 	}
 	return ret;
 }
+
+int btrfs_record_file_extent_comp(struct btrfs_trans_handle *trans,
+				  struct btrfs_root *root, u64 objectid,
+				  struct btrfs_inode_item *inode,
+				  u64 file_pos, u64 disk_bytenr,
+				  u64 extent_num_bytes, u64 ram_bytes,
+				  enum btrfs_compression_type comp)
```

Comment:
Instead of the longer and longer parameter list, what about something just like `btrfs_ordered_extent` inside the kernel, to record all the needed info?

So that we do not need two different `btrfs_record_file_extent()`, but a single one to handle them all, just like kernel.

This will need a dedicated cleanup patch first, to introduce some structure like `btrfs_ordered_extent`.